actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -1013,9 +1013,14 @@ def set_default_disable_with(value, tag_options)
         end
 
         def convert_direct_upload_option_to_url(options)
-          if options.delete(:direct_upload) && respond_to?(:rails_direct_uploads_url)
+          return options unless options.delete(:direct_upload)
+
+          if respond_to?(:main_app) && main_app.respond_to?(:rails_direct_uploads_url)
```

Comment:
In the context of just Action View, the `main_app` helper does not exist, hence these conditionals that check for its existence. The `elsif` clause is there to satisfy the tests for Action View only.

If there's a better way to do this, I'm all ears!

Reply:
Wouldn't it make more sense to check for `rails_direct_uploads_url` first, and then fallback to `main_app. rails_direct_uploads_url`?

First reason is that it allows you to customize that route in your engine if you wish.

Second reason is that in the vast majority of cases this code won't be executed from an engine, so might as well try the fast path first.

Reply:
Fixed!